main/main.c

```diff
@@ -1,548 +1,193 @@
-    //     .mode=0,                                //SPI mode 0
-    //     .spics_io_num=PIN_NUM_CAN_CS,           //CS pin
-    //     .queue_size=7,                          //We want to be able to queue 7 transactions at a time
-    //     .pre_cb=NULL,  //Specify pre-transfer callback to handle D/C line
-    // };
+    // sys_delay_ms(10000);
 
-    ESP_ERROR_CHECK(spi_bus_initialize(CANSPI_HOST, &buscfg, SPI_DMA_CH_AUTO));
-    // ESP_ERROR_CHECK(spi_bus_add_device(CANSPI_HOST, &devcfgCAN, &spi));
+    // Create default event loop needed by the  main app
+    ESP_ERROR_CHECK(esp_event_loop_create_default());
     
-    MCP251XFD MCP251863_CAN1 = DEFAULT_MCP251863_DRIVER_CONFIG(&spi);
-    MCP251XFD_Config MCP251863_CAN1_conf = DEFAULT_MCP251863_CONTROLLER_CONFIG(&MCP251863_SYSCLK1, &MCP251863_BTStats);
-    MCP251863_CAN1.DriverConfig |= MCP251XFD_DRIVER_INIT_SET_RAM_AT_0;
-
-    MCP251XFD_RAMInfos MCP251863_CAN1_FIFO_RAMInfos[MCP251863_FIFO_COUNT];
-    MCP251XFD_FIFO MCP251863_CAN1_FIFO_Conf[MCP251863_FIFO_COUNT] = {
-        { 
-            .Name = MCP251XFD_FIFO1, 
-            .Size = MCP251XFD_FIFO_16_MESSAGE_DEEP, 
-            .Payload = MCP251XFD_PAYLOAD_8BYTE,
-            .Direction = MCP251XFD_RECEIVE_FIFO, 
-            .ControlFlags = MCP251XFD_FIFO_NO_TIMESTAMP_ON_RX,
-            .InterruptFlags = MCP251XFD_FIFO_OVERFLOW_INT + MCP251XFD_FIFO_RECEIVE_FIFO_NOT_EMPTY_INT,
-            .RAMInfos = &MCP251863_CAN1_FIFO_RAMInfos[0], 
-        }, // SID: 0x000..0x1FF ; No EID
-        { 
-            .Name = MCP251XFD_FIFO2, 
-            .Size = MCP251XFD_FIFO_4_MESSAGE_DEEP, 
-            .Payload = MCP251XFD_PAYLOAD_8BYTE,
-            .Direction = MCP251XFD_TRANSMIT_FIFO, 
-            .Attempts = MCP251XFD_THREE_ATTEMPTS,
-            .Priority = MCP251XFD_MESSAGE_TX_PRIORITY16, 
-            .ControlFlags = MCP251XFD_FIFO_NO_RTR_RESPONSE,
-            .InterruptFlags = MCP251XFD_FIFO_TX_ATTEMPTS_EXHAUSTED_INT + MCP251XFD_FIFO_TRANSMIT_FIFO_NOT_FULL_INT,
-            .RAMInfos = &MCP251863_CAN1_FIFO_RAMInfos[1], 
-        },
-    };
-
-    //setup MCP25863
-    eERRORRESULT ret;
-    ret = MCP251863DeviceSetup(&MCP251863_CAN1, &MCP251863_CAN1_conf, MCP251863_CAN1_FIFO_Conf, MCP251863_FIFO_COUNT);
-
-    //Get MCP25863 device info 
-    eMCP251XFD_Devices device;
-    uint8_t deviceId;
-    uint8_t deviceRev;
-    ret = MCP251XFD_GetDeviceID(&MCP251863_CAN1, &device, &deviceId, &deviceRev);
-    ESP_LOGI("MCP INIT","Device ID result: %d", ret);
-    ESP_LOGI("MCP INIT","Device %d, ID %x, REV %x", device, deviceId, deviceRev);
-
-    const char* base_path = "/data";
-    const char* canLogPath = "/data/CANLogs";
-    sdmmc_card_t card;
-
-    int r = mountSDCard(base_path, &card);
-    ESP_LOGI("SD_CARD", "SD card initialized with status %d", r);
-    if(!r){
-        sdmmc_card_print_info(stdout, &card);
-    }
+    // vTaskDelay(pdMS_TO_TICKS(1000)); //delay for a bit to give time for debugger to attach
```

Comment:
Is this not actually necessary? Or should it be commented in whenever debugging is being done?

Reply:
Just helpful when the board is repeatedly crashing. Added a comment ot clarify.